tests/integration/utils2.go

```diff
@@ -131,59 +148,57 @@ var previousTestCaseTestName string
 func init() {
 	// We use environment variables instead of flags `go test ./...` throws for all packages
 	//  that don't have the flag defined
-	badgerFileValue, _ := os.LookupEnv(fileBadgerEnvName)
-	badgerInMemoryValue, _ := os.LookupEnv(memoryBadgerEnvName)
-	databaseDir, _ = os.LookupEnv(fileBadgerPathEnvName)
-	rootDatabaseDir, _ = os.LookupEnv(rootDBFilePathEnvName)
-	detectDbChangesValue, _ := os.LookupEnv(detectDbChangesEnvName)
-	inMemoryStoreValue, _ := os.LookupEnv(inMemoryEnvName)
-	repositoryValue, repositorySpecified := os.LookupEnv(repositoryEnvName)
-	setupOnlyValue, _ := os.LookupEnv(setupOnlyEnvName)
-	targetBranchValue, targetBranchSpecified := os.LookupEnv(targetBranchEnvName)
-	mutType, mutationTypeSpecified := os.LookupEnv(mutationTypeEnvName)
-
-	badgerFile = getBool(badgerFileValue)
-	badgerInMemory = getBool(badgerInMemoryValue)
-	inMemoryStore = getBool(inMemoryStoreValue)
-	DetectDbChanges = getBool(detectDbChangesValue)
-	SetupOnly = getBool(setupOnlyValue)
-
-	if !repositorySpecified {
+	httpClient, _ = strconv.ParseBool(os.Getenv(clientHttpEnvName))
+	goClient, _ = strconv.ParseBool(os.Getenv(clientGoEnvName))
+	badgerFile, _ = strconv.ParseBool(os.Getenv(fileBadgerEnvName))
+	badgerInMemory, _ = strconv.ParseBool(os.Getenv(memoryBadgerEnvName))
+	inMemoryStore, _ = strconv.ParseBool(os.Getenv(inMemoryEnvName))
+	DetectDbChanges, _ = strconv.ParseBool(os.Getenv(detectDbChangesEnvName))
+	SetupOnly, _ = strconv.ParseBool(os.Getenv(setupOnlyEnvName))
+
+	var repositoryValue string
+	if value, ok := os.LookupEnv(repositoryEnvName); ok {
+		repositoryValue = value
+	} else {
 		repositoryValue = "https://github.com/sourcenetwork/defradb.git"
 	}
 
-	if !targetBranchSpecified {
+	var targetBranchValue string
+	if value, ok := os.LookupEnv(targetBranchEnvName); ok {
+		targetBranchValue = value
+	} else {
 		targetBranchValue = "develop"
 	}
 
-	if mutationTypeSpecified {
-		mutationType = MutationType(mutType)
+	if value, ok := os.LookupEnv(mutationTypeEnvName); ok {
+		mutationType = MutationType(value)
 	} else {
 		// Default to testing mutations via Collection.Save - it should be simpler and
 		// faster. We assume this is desirable when not explicitly testing any particular
 		// mutation type.
 		mutationType = CollectionSaveMutationType
 	}
 
-	// default is to run against all
-	if !badgerInMemory && !badgerFile && !inMemoryStore && !DetectDbChanges {
-		badgerInMemory = true
-		// Testing against the file system is off by default
-		badgerFile = false
-		inMemoryStore = true
-	}
-
-	if DetectDbChanges {
+	// Set default values for the specified testing mode.
+	switch {
+	case DetectDbChanges:
+		// Change detector runs using only the go client type.
```

Comment:
todo: I think the code should change.  Whilst we currently think that running the change detector via the http client is pointless, it is not technically impossible and we shouldnt prevent it.  The environment variables should still be respected, and it will cause confusion (and bugs) if they are not.

Can you please instead change the `make test:changes` action to instead explicitly run using just the Go client.  Regardless of what the default clients are (see other comment about defaults).